### PR TITLE
Master account invoice report mep

### PR DESCRIPTION
**Pull request by mep-odoo**
Task : https://www.odoo.com/web#id=30501&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

Pad : https://pad.odoo.com/p/r.4c5e2a8c432e737fde588c32fdf2e769

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/report_invoice.xml [hunk 1]
```diff
@@ -99,55 +99,46 @@
...
-                                    </tr>
-                                </thead>
-                                <tbody>
-                                    <tr t-foreach="o.tax_line_ids" t-as="t">
-                                        <td><span t-field="t.tax_id.description"/></td>
-                                        <td class="text-right">
-                                            <span t-field="t.base" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
-                                        </td>
+                                    <t t-set="tax_groups_length" t-value="len(o._get_tax_amount_by_group())"/>
+                                    <t t-foreach="o._get_tax_amount_by_group()" t-as="amount_by_group">
```
Comment (smetl) [inline addons/account/views/report_invoice.xml]: Don't call `_get_tax_amount_by_group()` multiple times. If you need to reuse this, store it in a variable like: 
`<t t-set="tax_groups" t-value="o._get_tax_amount_by_group()"/>`
`<t t-foreach="tax_groups" t-as="amount_by_group">`

and to retrieve the len: `len(tax_groups)`.

I also notice you never use `tax_groups_length`... is it an old stuff?

File: addons/l10n_ch/data/account_data.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <record id="tax_group_tva_38" model="account.tax.group">
+            <field name="name">TVA 3.8%</field>
+        </record>
+        <record id="tax_group_tva_8" model="account.tax.group">
+            <field name="name">TVA 8%</field>
+        </record>
+
+    </data>
+</odoo>
+
```
Comment (smetl) [inline addons/l10n_ch/data/account_data.xml]: remove empty space

File: addons/sale/report/sale_report_templates.xml [hunk 3]
```diff
@@ -136,33 +136,47 @@
...
-                        <tr class="border-black">
-                            <td><strong>Total</strong></td>
-                            <td class="text-right">
-                                <span t-field="doc.amount_total"
-                                    t-options='{"widget": "monetary", "display_currency": doc.pricelist_id.currency_id}'/>
-                            </td>
-                        </tr>
-                    </table>
+                            <t t-set="tax_groups_length" t-value="len(doc._get_tax_amount_by_group())"/>
+                            <t t-foreach="doc._get_tax_amount_by_group()" t-as="amount_by_group">
```
Comment (smetl) [inline addons/sale/report/sale_report_templates.xml]: same here, multiple `_get_tax_amount_by_group()` calls.

File: addons/sale/models/sale.py [hunk 4]
```diff
@@ -546,18 +546,19 @@ def _get_tax_amount_by_group(self):
...
                 for t in taxes:
-                    res[group] += t['amount']
+                    res[group]['amount'] += t['amount']
+                    res[group]['base'] += t['base']
                 if tax.include_base_amount:
                     base_tax += tax.compute_all(price_reduce + base_tax, quantity=1, product=line.product_id,
                                                 partner=self.partner_shipping_id)['taxes'][0]['amount']
         res = sorted(pycompat.items(res), key=lambda l: l[0].sequence)
-        res = [(l[0].name, l[1]) for l in res]
+        res = [(l[0].name, l[1]['amount'], l[1]['base'], len(res)) for l in res]
```
Comment (smetl) [inline addons/sale/models/sale.py]: Why adding the `len(res)` on each line?

Comment (mep-odoo) [inline addons/sale/models/sale.py]: @smetl 
actually, task goal is to display only tax group name + tax amount when invoice/sale order is having only one tax group. And if invoice/sale order is having multiple tax groups then it needs to show tax groups names + base amount + tax amount.
So, in account.invoice we have a tax_line_ids field, so we can use len() like https://github.com/odoo/odoo/pull/16724/commits/e224d0b4fe679e005d215a63b755e65323c973f1#diff-774f4f512fb11d642b30d54aa5980f6bR115

But in sale.order/sale.order.line model we do not have such field, so to find the tax groups count I have to pass the length of tax groups in this function and I checked it in template like https://github.com/odoo/odoo/pull/16724/commits/90ae1dc7bbf5866e5901218bd746532d0fa8aaa5#diff-e7cfd97817c83d4d82652ff1ddcf2d9eR153.

Comment (smetl) [inline addons/sale/models/sale.py]: ok

**Pull request merged**